test/e2e_node/cpu_manager_test.go

```diff
@@ -0,0 +1,424 @@
+				cpuRequest: "1000m",
+				cpuLimit:   "1000m",
+			},
+		}
+		pod = makeCPUManagerPod("gu-pod", ctnAttrs)
+		pod = f.PodClient().CreateSync(pod)
+
+		By("checking if the expected cpuset was assigned")
+		cpu1, cpu2 = 1, 2
+		if isHTEnabled() {
+			cpuList = cpuset.MustParse(getCPUSiblingList(0)).ToSlice()
+			if cpuList[1] != 1 {
+				cpu1, cpu2 = cpuList[1], 1
+			}
+		}
+
+		expAllowedCPUsListRegex = fmt.Sprintf("^%d|%d\n$", cpu1, cpu2)
+		err = f.PodClient().MatchContainerOutput(pod.Name, pod.Spec.Containers[0].Name, expAllowedCPUsListRegex)
+		framework.ExpectNoError(err, "expected log not found in container [%s] of pod [%s]",
+			pod.Spec.Containers[0].Name, pod.Name)
+
+		err = f.PodClient().MatchContainerOutput(pod.Name, pod.Spec.Containers[0].Name, expAllowedCPUsListRegex)
+		framework.ExpectNoError(err, "expected log not found in container [%s] of pod [%s]",
+			pod.Spec.Containers[1].Name, pod.Name)
+
+		By("by deleting the pods and waiting for container removal")
+		deletePods(f, []string{pod.Name})
+		waitForContainerRemoval(fmt.Sprintf("%s_%s", pod.Spec.Containers[0].Name, pod.Name))
+		waitForContainerRemoval(fmt.Sprintf("%s_%s", pod.Spec.Containers[1].Name, pod.Name))
+
+		By("running multiple Gu pods")
+		ctnAttrs = []ctnAttribute{
+			{
+				ctnName:    "gu-container1",
+				cpuRequest: "1000m",
+				cpuLimit:   "1000m",
+			},
+		}
+		pod1 = makeCPUManagerPod("gu-pod1", ctnAttrs)
+		pod1 = f.PodClient().CreateSync(pod1)
+
+		ctnAttrs = []ctnAttribute{
+			{
+				ctnName:    "gu-container2",
+				cpuRequest: "1000m",
+				cpuLimit:   "1000m",
+			},
+		}
+		pod2 = makeCPUManagerPod("gu-pod2", ctnAttrs)
+		pod2 = f.PodClient().CreateSync(pod2)
+
+		By("checking if the expected cpuset was assigned")
+		cpu1, cpu2 = 1, 2
+		if isHTEnabled() {
+			cpuList = cpuset.MustParse(getCPUSiblingList(0)).ToSlice()
+			if cpuList[1] != 1 {
+				cpu1, cpu2 = cpuList[1], 1
+			}
+		}
+
+		expAllowedCPUsListRegex = fmt.Sprintf("^%d\n$", cpu1)
+		err = f.PodClient().MatchContainerOutput(pod1.Name, pod1.Spec.Containers[0].Name, expAllowedCPUsListRegex)
+		framework.ExpectNoError(err, "expected log not found in container [%s] of pod [%s]",
+			pod1.Spec.Containers[0].Name, pod1.Name)
+
+		expAllowedCPUsListRegex = fmt.Sprintf("^%d\n$", cpu2)
+		err = f.PodClient().MatchContainerOutput(pod2.Name, pod2.Spec.Containers[0].Name, expAllowedCPUsListRegex)
+		framework.ExpectNoError(err, "expected log not found in container [%s] of pod [%s]",
+			pod2.Spec.Containers[0].Name, pod2.Name)
+
+		By("by deleting the pods and waiting for container removal")
+		deletePods(f, []string{pod1.Name, pod2.Name})
+		waitForContainerRemoval(fmt.Sprintf("%s_%s", pod1.Spec.Containers[0].Name, pod1.Name))
+		waitForContainerRemoval(fmt.Sprintf("%s_%s", pod2.Spec.Containers[0].Name, pod2.Name))
+
+		setOldKubeletConfig(f)
+	})
+}
+
+// Serial because the test updates kubelet configuration.
+var _ = framework.KubeDescribe("CPU Manager [Serial]", func() {
```

Comment:
Is the `[Serial]` suffix still required now that we moved everything into one test block?

Reply:
Yes, I think so. Serial and/or Disruptive is being used in all the node e2e tests which change the kubelet config. Also, serial tests are run separately (https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/jenkins/image-config.yaml vs https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/jenkins/image-config.yaml).

Reply:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/jenkins/jenkins-serial-ubuntu.properties#L9 is also required.

Reply:
OK, that makes sense.